base.py

```diff
@@ -69,3 +76,41 @@ def _validate_input(self, X: List):
 
         X = [MolFromSmiles(x) if type(x) == str else x for x in X]
         return X
+
+
+class FingerprintGeneratorInterface(ABC):
```

Comment:
Maybe `Mixin` instead of `Interface`, since this is, well, a mixin. And also Scikit-learn uses that convention

Reply:
Done